internal/mesh/internal/controllers/sidecarproxy/controller_test.go

```diff
@@ -706,7 +706,7 @@ func (suite *controllerTestSuite) TestController() {
 			// We should also get a new web proxy template resource as this destination should be added again.
 			webProxyStateTemplate = suite.client.WaitForNewVersion(suite.T(), webProxyStateTemplateID, webProxyStateTemplate.Version)
```

Comment:
We could probably remove some of these explicit version change waits since they're lossing compared to the state check, but I thought it might be informative to keep them in case no update is made at all.

Reply:
From a quick scan of the code I don't think `webProxyStateTemplate` is actually used any more. Would you feel comfortable deleting them? The (small) risk is that they get copy-pasted into other tests without being fully understood.

Reply:
I think it's still being used in our version checks (`webProxyStateTemplate.Version` should change each time we capture a new value for it), as well as one call to `assertTrafficPermissionDefaultPolicy` - unless I'm confusing something it looks like it probably needs to stick around for now